inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/flink/deserialization/DeserializationSchema.java

```diff
@@ -122,7 +145,23 @@ public void processElement(
                     context.output(METRIC_DATA_OUTPUT_TAG, metricData);
                 }
 
-                collector.collect(recordTransformer.toSerializedRecord(sinkRecord));
+                SerializedRecord serializedSinkRecord = recordTransformer.toSerializedRecord(sinkRecord);
+
+                if (auditImp != null) {
+                    Pair<String, String> groupIdAndStreamId = inLongGroupIdAndStreamIdMap.getOrDefault(
+                            serializedRecord.getDataFlowId(),
+                            Pair.of("", ""));
+
+                    auditImp.add(
+                            Constants.METRIC_AUDIT_ID_FOR_INPUT,
+                            groupIdAndStreamId.getLeft(),
+                            groupIdAndStreamId.getRight(),
+                            sinkRecord.getTimestampMillis(),
```

Comment:
Please check that sinkRecord.getTimestampMillis() is the generated time of pulsar/tube Message or the logged time of user data or current time.
    public DeserializationResult<SerializedRecord> deserialize(@SuppressWarnings("rawtypes") Message message)
            throws IOException {
        final byte[] data = message.getData();
        return DeserializationResult.of(new SerializedRecord(dataFlowId, message.getEventTime(), data), data.length);
    }
            deserializer.flatMap(mixedRow, new CallbackCollector<>((row -> {
                // each tid might be associated with multiple data flows
                for (long dataFlowId : dataFlowIds) {
                    collector.collect(new Record(dataFlowId, System.currentTimeMillis(), row));
                }
            })));

Reply:
sinkRecord.getTimestampMillis() is event time extracted from pulsar/tube message

Reply:
It is different between the generated time of pulsar/tube Message,the logged time of user data, current time.
current time>the generated time of pulsar/tube Message>the logged time of user data in agent sdk.
The log time of audit interface is for calculating delay interval in every step.